optimum/intel/openvino/modeling_base_seq2seq.py

```diff
@@ -232,8 +234,6 @@ def _from_pretrained(
         # Load model from hub
         else:
             model_file_names = {"encoder": encoder_file_name, "decoder": decoder_file_name}
-            if use_cache:
```

Comment:
one suggestion would be to follow the same pattern you used to check for statefulness when loading from local files based on existence/absence of `decoder_with_past_file_name`, we can check here as well using
`use_cache and decoder_with_past_file_name in repo_files ` with `repo_files = huggingface_hub.list_repo_files(model_name_or_path, revision=revision, token=token, ...)`
avoiding changes to the onnx path and the subsequent changes.

Reply:
file existance not always means that it should be loaded, sometimes people save model multiple time in the same directory (so if previous run uses stateless and current run stateful) it will be incorrect

Reply:
Totally agree with that, I'm only suggesting it because you already used this "file existence/absence" in the local files loading path (right above this one).

Reply:
improved from_onnx usage and conditions for loading decoder_with_past